pkg/controller/operators/olm/operatorgroup.go

```diff
@@ -64,8 +65,51 @@ func (a *Operator) syncOperatorGroups(obj interface{}) error {
 		"namespace":     op.GetNamespace(),
 	})
 
+	// Query OG in this namespace
+	groups, err := a.lister.OperatorsV1().OperatorGroupLister().OperatorGroups(op.GetNamespace()).List(labels.Everything())
+	if err != nil {
+		logger.WithError(err).Warnf("failed to list OperatorGroups in the namespace")
+	}
+
+	// Check if there is a stale multiple OG condition and clear it if existed.
+	if len(groups) == 1 {
+		og := groups[0]
+		if c := meta.FindStatusCondition(og.Status.Conditions, v1.MutlipleOperatorGroupCondition); c != nil {
+			meta.RemoveStatusCondition(&og.Status.Conditions, v1.MutlipleOperatorGroupCondition)
+			if og.GetName() == op.GetName() {
```

Comment:
It seems like we may want to exit early if the provided operatorGroup does not share a name with the found operatorGroup. Currently this code will have hit an error attempting to update an operatorGroup that does not exist.

Reply:
NVM - this seems to be fine.